src/scipp/core/binning.py

```diff
@@ -0,0 +1,99 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+from typing import Dict, List, Optional, Union
+
+from .._scipp import core as _cpp
+from .variable import array, Variable, linspace, arange
+from .bins import bin, histogram
+
+
+def _get_coord(x, name):
+    event_coord = x.bins.meta.get(name) if x.bins is not None else None
+    return x.meta.get(name, event_coord)
```

Comment:
I think this is what `sc.bin` does: It "trusts" the info given by the bin coords. If we would not do that, `sc.bin` becomes computationally more complex, since essentially it would need to bin from scratch every time.